src/detect-detection-filter.c

```diff
@@ -304,12 +305,10 @@ static int DetectDetectionFilterTestParse02 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track both,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(df);
+    DetectDetectionFilterFree(NULL, df);
```

Comment:
we can only get here if `df` is NULL, so this free statement is redundant

Reply:
Ok . Thank you for reviews!